### ISSUE TITLE
Group still assigned to agent when it's been removed

### ISSUE THREAD

**Issue by DaveVG1** [wazuh/wazuh#2028]
## Bug report

**OS**
_CentOS7_

**Wazuh version**
_3.7.1-3724_

**Install type**
_manager_

**Install method**
_packages_

**Python version**
_Python 3.6.3_

**Log sample**

```
[root@vm-centos7 Group1]# agent_groups -a -g Group1
Do you want to create the group 'Group1'? [y/N]: y
Group 'Group1' created.

[root@vm-centos7 Group1]# agent_groups -a -i 001 -g Group1
Do you want to add the group 'Group1' to the agent '001'? [y/N]: y
Group 'Group1' added to agent '001'.
[root@vm-centos7 Group1]# agent_groups -r -i 001 -g default
Do you want to delete the group 'default' of agent '001'? [y/N]: y
Group 'default' unset for agent '001'.
[root@vm-centos7 Group1]# agent_groups -l
Groups (2):
  default (1)
  Group1 (1)
Unassigned agents: 0.

[root@vm-centos7 Group1]# agent_groups -r -g Group1
Do you want to remove the 'Group1' group? [y/N]: y
All selected groups were removed
Affected agents: 001.
[root@vm-centos7 Group1]# agent_groups -l
Groups (1):
  default (1)
Unassigned agents: 0.

[root@vm-centos7 Group1]# cat /var/ossec/queue/agent-groups/001
Group1
```

Steps
----------------
1. Create a cluster configuration (one master and one worker at least).
2. Create an agent and point it to the worker.
3. Create a new group and assign it to the agent. Delete default group from the agent.
4. Delete the group created and check (```cat /var/ossec/queue/agent-groups/AGENT_ID```, ```agent_groups -l```, etc.)

Bug description
----------------
When a group (```Group1```) is removed (which had agent ```001``` assigned to it), the agent doesn't go to unassigned agents. In fact, it still has the same group assigned to it (even when the agent is restarted)

**Comment by vikman90** [comment]
Possibly fixed here: #1863

**Comment by chemamartinez** [comment]
Hi,

Let me close this issue since PR #1863 adds a totally different way to manage groups assignation and this behavior has not been detected anymore.

Regards.
